tests/SAML2/XML/spid/IssuerTest.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SAML2\XML\spid;
+
+use InvalidArgumentException;
+use PHPUnit\Framework\TestCase;
+use SAML2\Constants;
+use SAML2\DOMDocumentFactory;
+use SAML2\Utils;
+
+/**
+ * Class \SAML2\XML\saml\IssuerTest
+ */
+class IssuerTest extends TestCase
+{
+    /**
+     * @return void
+     */
+    public function testMarshalling(): void
+    {
+        $issuer = new Issuer('TheIssuerValue', Constants::NAMEID_ENTITY, 'TheSPProvidedID', 'TheNameQualifier', 'TheSPNameQualifier');
+        $issuerElement = $issuer->toXML();
+        $issuerElements = Utils::xpQuery($issuerElement, '/saml_assertion:Issuer');
```

Comment:
I find it easier to have an XML document built from a string (which can be done in `setUp()`, and then compare that one with the string generated by the object we just created. That way we don't need to be manipulating the DOM in order to test if it's correct. Actually, such comparison allows us to verify that what we generate is exactly what we expect, while with this approach we could be missing things because the single tests we are doing are fine.

An example from today: we had a bug in one class (I think `SPSSODescriptor`) that made the `md:Extensions` object be duplicated into the resulting DOM. Using an approach like this, I wouldn't have noticed the problem. However, comparing the exact resulting string uncovered it.